test/integration/puppetlabs/services/certificate_authority/certificate_authority_int_test.clj

```diff
@@ -167,6 +169,40 @@
                 (is (= 403 (:status response))
                     (ks/pprint-to-string response))))))))))
 
+(deftest ^:integration certificate-with-ca-true-extension-refused
+  (testing (str "Validates that the server rejects a csr for signing"
+                " that has the v3 CA:TRUE extension")
+    (let [master-conf-dir (str test-resources-dir 
+                               "/ca_true_test/master/conf")]
+      (let [key-pair  (ssl-utils/generate-key-pair)
+            subjectDN (ssl-utils/cn "test_cert_ca_true")
+            serial 1
+            public-key (ssl-utils/get-public-key key-pair)
+            ca-ext    (ca/create-ca-extensions subjectDN
+                                               serial
+                                               public-key)
+            csr       (ssl-utils/generate-certificate-request key-pair
+                                                              subjectDN
+                                                              ca-ext)]
+            (ssl-utils/obj->pem! csr (str master-conf-dir 
+                                          "/ssl/ca/requests"
+                                          "/test_cert_ca_true.pem"))
```

Comment:
I think the test will also need to clean this up at the end, because otherwise it looks like it's left behind. Often when writing tests like this we'll have them output files to `/tmp` or `target` to not have to worry so much, but I don't know if that would be possible here since we're using the `master-conf-dir`.